### PR TITLE
Tests for sum and average.

### PR DESCRIPTION
**Pull request by MarkDuckworth**
Tests for sum and average, plus a few fixes.

### CODE REVIEW

**Review by ehsannas — COMMENTED** [review]
LGTM. Thanks for the thorough test coverage. Maybe we should consider whether it's necessary to allow special characters and whether `TrimBackticks` is needed?

File: packages/firestore/test/lite/integration.test.ts [hunk 1]
```diff
@@ -2651,10 +2653,10 @@ describe('Aggregate quries', () => {
...
   it('aggregateQuerySnapshotEqual on different queries be falsy', () => {
     const testDocs = [
-      { author: 'authorA', title: 'titleA' },
-      { author: 'authorA', title: 'titleB' },
-      { author: 'authorB', title: 'titleC' },
-      { author: 'authorB', title: 'titleD' }
+      { author: 'authorA', title: 'titleA', rating: 1 },
+      { author: 'authorA', title: 'titleB', rating: 5 },
+      { author: 'authorB', title: 'titleC', rating: 4 },
+      { author: 'authorB', title: 'titleD', rating: 3 }
```
[the remark this answers is not in the document]
Comment (MarkDuckworth) [reply packages/firestore/test/lite/integration.test.ts:2659]: Removed.

File: packages/firestore/test/lite/integration.test.ts [hunk 2]
```diff
@@ -2712,3 +2714,141 @@ describe('Aggregate quries', () => {
...
+    ];
+    return withTestCollectionAndInitialData(testDocs, async coll => {
+      const query1 = query(coll, where('author', '==', 'authorA'));
+      const snapshot1 = await getAggregate(query1, { sum: sum('rating') });
+      const snapshot2 = await getAggregate(query1, { avg: average('rating') });
+
+      // @ts-expect-error
```
[the remark this answers is not in the document]
Comment (MarkDuckworth) [reply packages/firestore/test/lite/integration.test.ts:2732]: `snapshot1` and `snapshot2` have different types and therefore the call to `aggregateQuerySnapshotEqual(...)` will cause a TS error. To test the `aggregateQuerySnapshotEqual` method for JS users, we ignore the TS error.

I added a code comment stating this too.

File: packages/firestore/test/lite/integration.test.ts [hunk 3]
```diff
@@ -2712,3 +2714,141 @@ describe('Aggregate quries', () => {
...
+    ];
+    return withTestCollectionAndInitialData(testDocs, async coll => {
+      const query1 = query(coll, where('author', '==', 'authorA'));
+      const snapshot1 = await getAggregate(query1, { foo: average('rating') });
+      const snapshot2 = await getAggregate(query1, { bar: average('rating') });
+
+      // @ts-expect-error
+      expect(aggregateQuerySnapshotEqual(snapshot1, snapshot2)).to.be.false;
+    });
+  });
```
[the remark this answers is not in the document]
Comment (MarkDuckworth) [reply packages/firestore/test/lite/integration.test.ts:2752]: `snapshot1` and `snapshot2` have different types and therefore the call to `aggregateQuerySnapshotEqual(...)` will cause a TS error. To test the `aggregateQuerySnapshotEqual` method for JS users, we ignore the TS error.

I added a code comment stating this too.

File: packages/firestore/test/lite/integration.test.ts [hunk 4]
```diff
@@ -2712,3 +2714,141 @@ describe('Aggregate quries', () => {
...
+    const testDocs = [
+      { author: 'authorA', title: 'titleA', rating: 1 },
+      { author: 'authorA', title: 'titleB', rating: 5 },
+      { author: 'authorB', title: 'titleC', rating: 4 },
+      { author: 'authorB', title: 'titleD', rating: 3 }
+    ];
+    return withTestCollectionAndInitialData(testDocs, async coll => {
+      const query1 = query(coll, where('author', '==', 'authorA'));
+      const snapshot1 = await getAggregate(query1, { foo: average('rating') });
+      const snapshot2 = await getAggregate(query1, { foo: average('rating') });
```
[the remark this answers is not in the document]
Comment (MarkDuckworth) [reply packages/firestore/test/lite/integration.test.ts:2764]: Agreed. I added your change.

File: packages/firestore/test/integration/api/aggregation.test.ts [hunk 5]
```diff
@@ -279,3 +324,1177 @@ apiDescribe('Aggregation quries', (persistence: boolean) => {
...
+          pages: 50,
+          year: 2020,
+          rating: Number.MAX_VALUE
+        }
+      };
+      return withTestCollection(persistence, testDocs, async coll => {
+        const snapshot = await getAggregateFromServer(coll, {
+          totalRating: sum('rating')
+        });
+        expect(snapshot.data().totalRating).to.equal(Number.POSITIVE_INFINITY);
```
[the remark this answers is not in the document]
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/aggregation.test.ts]: That was not my experience. The server rounded `MAX_VALUE + 1` down to `MAX_VALUE` as `+1`. Double precision at that magnitude would require a value of approximately `+ 1e293` to increment the accumulator.

File: packages/firestore/test/integration/api/aggregation.test.ts [hunk 6]
```diff
@@ -279,3 +324,1177 @@ apiDescribe('Aggregation quries', (persistence: boolean) => {
...
+          pages: 150,
+          year: 2021,
+          rating: 10
+        }
+      };
+      return withTestCollection(persistence, testDocs, async coll => {
+        const snapshot = await getAggregateFromServer(coll, {
+          averageRating: average('rating')
+        });
+        expect(snapshot.data().averageRating).to.equal(9.5);
```
[the remark this answers is not in the document]
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/aggregation.test.ts]: Updated. This scenario was failing on the backend and never hit this assertion.

File: packages/firestore/test/integration/api/aggregation.test.ts [hunk 7]
```diff
@@ -279,3 +324,1177 @@ apiDescribe('Aggregation quries', (persistence: boolean) => {
...
+          pages: 150,
+          year: 2021,
+          rating: 10
+        }
+      };
+      return withTestCollection(persistence, testDocs, async coll => {
+        const snapshot = await getAggregateFromServer(coll, {
+          averageRating: average('rating')
+        });
+        expect(snapshot.data().averageRating).to.equal(10);
```
[the remark this answers is not in the document]
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/aggregation.test.ts]: Thanks. Fixed

File: packages/firestore/test/lite/integration.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ehsannas) [reply packages/firestore/test/lite/integration.test.ts:2735]: Thanks

File: packages/firestore/test/integration/api/aggregation.test.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (ehsannas) [reply packages/firestore/test/integration/api/aggregation.test.ts:889]: I see, thanks

**Review by ehsannas — APPROVED** [review]
LGTM

**Pull request merged**